tests/units/classes/lizmapServicesTest.php

```diff
@@ -0,0 +1,291 @@
+		$testModify3_1 = array(
+			'undefined' => 'test',
+		);
+		return array(
+			array($testModify1, $testModify1_1, null, null, true),
+			array($testModify1, $testModify2_1, 'adminSenderEmail', 'test.test@3liz.org', true),
+			array($testModify3, $testModify1_1, null, null, true),
+			array($testModify1, null, null, null, false),
+			array(null, $testModify1_1, 'adminSenderEmail', 'test.test@test.com', true),
+			array(null, $testModify3_1, null, null, false)
+		);
+	}
+
+	/**
+	 * @dataProvider getModifyGlobalData
+	 */
+
+	public function	testModifyGlobal($globalConfig, $newConfig, $changedProperty, $changedValue, $expectedReturnValue)	{
+
+		$testLizmapServices = new LizmapServices(array(), (object)$globalConfig, false, '');
+		$this->assertEquals($expectedReturnValue, $testLizmapServices->modify($newConfig));
+		if (isset($changedProperty))	{
+			$this->assertEquals($changedValue, $testLizmapServices->$changedProperty);
+		}
+		unset($testLizmapServices);
+	}
+
+	public function getModifyLocalData()	{
+		$testModify1 = array(
+			'services' => array(
+				'appName' => 'Lizmap',
+				'adminContactEmail' => 'test.test@test.com')
+		);
+		$testModify2 = array(
+			'test' => 'test'
+		);
+		$testModify1_1 = array(
+			'appName' => 'Lizmap',
+			'adminContactEmail' => 'test.test@test.com'
+		);
+		$testModify2_1 = array(
+			'appName' => 'Lizmap',
+			'adminContactEmail' => 'test.test@3liz.org'
+		);
+		$testModify3_1 = array(
+			'appName' => 'Lizmap',
+			'adminContactEmail' => 'test.test@test.com',
+			'debugMode' => 'off'
+		);
+		return array(
+			array($testModify1, $testModify1_1, null, null, true),
+			array($testModify1, $testModify2_1, 'adminContactEmail', 'test.test@3liz.org', true),
+			array($testModify1, $testModify3_1, 'debugMode', 'off', true),
+			array($testModify1, null, null, null, false),
+			array($testModify2, $testModify1_1, 'adminContactEmail', 'test.test@test.com', true),
+			array(null, $testModify1_1, 'adminContactEmail', 'test.test@test.com', true),
+		);
+	}
+
+	/**
+	 * @dataProvider getModifyLocalData
+	 */
+
+	public function testModifyLocal($localConfig, $newConfig, $changedProperty, $changedValue, $expectedReturnValue)	{
+
+		$testLizmapServices = new LizmapServices($localConfig, null, false, '');
+		$this->assertEquals($expectedReturnValue, $testLizmapServices->modify($newConfig));
+		if (isset($changedProperty))	{
+			$this->assertEquals($changedValue, $testLizmapServices->$changedProperty);
+		}
+		unset($testLizmapServices);
+	}
+
+	public function getSaveIntoIniData()	{
+		$ini1 = array(
+			'appName' => 'Lizmap',
+			'adminContactEmail' => 'test.test@test.com'
+		);
+		$ini1_1 = array(
+			'appName' => 'Lizmap'
+		);
+		$ini2 = array();
+		$liveIni = array(
+			'webmasterEmail' => 'test.test@test.com',
+			'webmasterName' => 'Adrien'
+		);
+		return array(
+			array('$testLizmapServices->appName = "Lizmap"; $testLizmapServices->adminContactEmail = "test.test@test.com";', $ini1, null, 'services', false),
```

Comment:
I don't like this very much. Using `eval` is considered as an evil function ;-). You should set an array here, instead.